tests/meta/transfer/test_external_data.py

```diff
@@ -60,38 +60,47 @@ def test_get_destination_path(
     assert destination_path == Path("/path/on/hasta/cust000/ADM1/")
 
 
-def test_download_sample(
-    apps_dir: Path,
-    customer_id: str,
-    cust_sample_id: str,
+def test_transfer_sample_files_from_source(
+    caplog,
+    cg_context: CGConfig,
+    customer_id,
+    cust_sample_id,
     external_data_api: ExternalDataAPI,
+    external_data_directory,
+    helpers,
     mocker,
-    sample_id: str,
-    ticket_nr: int,
+    sample_store,
+    ticket_nr,
 ):
-    """Test for transferring external data via SLURM"""
+    caplog.set_level(logging.INFO)
 
-    # GIVEN paths needed to run rsync
-    mocker.patch.object(ExternalDataAPI, "create_log_dir")
-    ExternalDataAPI.create_log_dir.return_value = apps_dir
+    # GIVEN a Store with three samples, where only two samples are present in the source folder
+    helpers.add_sample(
+        store=external_data_api.status_db, name=cust_sample_id + "1", ticket=ticket_nr
+    )
+    helpers.add_sample(
+        store=external_data_api.status_db, name=cust_sample_id + "2", ticket=ticket_nr
+    )
+    helpers.add_sample(
+        store=external_data_api.status_db, name=cust_sample_id + "3", ticket=ticket_nr
+    )
+
+    mocker.patch.object(Store, "get_customer_id_from_ticket")
+    Store.get_customer_id_from_ticket.return_value = customer_id
 
     mocker.patch.object(ExternalDataAPI, "get_source_path")
-    ExternalDataAPI.get_source_path.return_value = apps_dir
+    external_data_api.get_source_path.return_value = Path("").joinpath(external_data_directory)
 
-    mocker.patch.object(ExternalDataAPI, "get_destination_path")
-    ExternalDataAPI.get_destination_path.return_value = apps_dir
+    external_data_api.caesar_path = str(Path("").joinpath(*external_data_directory.parts[:-2]))
+    external_data_api.hasta_path = str(Path("").joinpath(*external_data_directory.parts[:-1], "%s"))
```

Comment:
```suggestion
    external_data_api.source_path = str(Path("").joinpath(*external_data_directory.parts[:-2]))
    external_data_api.destination_path = str(Path("").joinpath(*external_data_directory.parts[:-1], "%s"))
```
Hasta and Caesar are not going to live forever - better to use agnostic names